### ISSUE TITLE
Spins forever on trying to transcode and play corrupt video

### ISSUE THREAD

**Issue by brianjmurrell** [jellyfin/jellyfin#2277]
**Describe the bug**
When a transcode fails, due to a corrupt video perhaps, the web-ui just spins and spins forever, never letting the user know to stop waiting because the transcode failed:

**To Reproduce**

1. Insert a corrupt video (one that ffmpeg cannot decode) into the library
1. Open web-ui to the corrupt video
1. Play video

**Expected behavior**
I should get an error message on the web-ui that the video could not be played.

**Logs**
```
[2020-01-13 21:21:36.059 -05:00] [INF] /usr/bin/ffmpeg -fflags +genpts -f mp4 -i file:"/video/[redacted].mp4" -map_metadata -1 -map_chapters -1 -threads 0 -sn -codec:v:0 copy -flags -global_header -vsync -1 -codec:a:0 libmp3lame -ac 2 -ab 192000  -f segment -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -segment_time 3  -individual_header_trailer 0 -segment_format mpegts -segment_list_entry_prefix "hls/ab34ef6e40815c5c174f02a84790cfe7/" -segment_list_type m3u8 -segment_start_number 0 -segment_list "/var/lib/jellyfin/transcoding-temp/ab34ef6e40815c5c174f02a84790cfe7.m3u8" -y "/var/lib/jellyfin/transcoding-temp/ab34ef6e40815c5c174f02a84790cfe7%d.ts"
[2020-01-13 21:21:36.174 -05:00] [ERR] FFMpeg exited with code 1
```

**System (please complete the following information):**
 - OS: Fedora 31
 - Browser: Chrome
 - Jellyfin Version: 10.4.2
 - Installed Plugins: N/A
 - Reverse proxy: no

**Comment by JustAMan** [comment]
@brianjmurrell could you please enable debug logging? It might shed some light here.

**Comment by Artiume** [comment]
Request the FFmpeg logs, that should shed some light on the transcoding error

**Comment by brianjmurrell** [comment]
@Artiume Not at all relevant.  The fact that there was an FFmpeg encoding error is not in dispute and completely not the point of this ticket.  The point is that FFmpeg encoding errors are going to happen (garbage in garbage out).

Jellyfin needs to be much more resilient and informative about it when it happens rather than just indefinitely spinning on the user's screen leaving the user in the dark about what happened.

@JustAMan Apologies for the lack of logs.  I just haven't had the opportunity yet.  Although, this should not at all be difficult to reproduce.  Or are you saying that Jellyfin should already handle this more gracefully than it does and it's just not in my case?

**Comment by Artiume** [comment]
Depending on how it fails in ffmpeg will tell us how to fix it in the code :)

**Comment by brianjmurrell** [comment]
@Artiume No.  I think you need to go reread the ticket description again because you are still **completely** missing the point.  How or why FFmpeg failed is completely irrelevant here.

> jellyfin can try an alternative decoding method.

The video was corrupt.  FFmpeg was entirely expected to fail, always.  This is the situation that needs to be handled.

The problem is that when it does fail (and it will -- corrupt videos exist), it leaves the user clueless about it, with a never ending spinner.  This never ending spinner and complete lack of any other indication (i.e. to the user trying to play the video -- checking logs doesn't count -- that is not a "user" operation -- my mother is not going to go check logs) that FFmpeg (or more generically playing the video) failed is the point of this ticket.

While I appreciate that you want to help and provide input, please try to keep the original point of the ticket in mind and don't be distracted as to **why** FFmpeg might have failed, but just that it did and that's what needs to be handled more gracefully.

**Comment by Artiume** [comment]
So it's all one in the same. The wrapper will work as such; the first if statement goes around the entire statement. If HWA was used, it attempts software conversion. If that fails, it either informs the user, plays whatever is available such as the container and audio only. Or it can exit the session. That's what the team and I are considering. The same issue happens constantly to me on LiveTV all the damn time so don't think I don't understand the situation. Currently the engine performs an infinite loop and never attempts to exit. So stop being an ass and let me help.

**Comment by brianjmurrell** [comment]
@Artiume The problem is that all of your problem suggestions/descriptions were focused on why ffmpeg failed:

> Request the FFmpeg logs, that should shed some light on the transcoding error

and

> We have to build a wrapper around StreamBuilder to watch the error streams from ffmpeg. That way jellyfin can try an alternative decoding method.

Both focus on the cause of the ffmpeg error and/or falling back to something else expected to make ffmpeg successful, which as your previous suggestions describe it, is irrelevant.  Nowhere in your previous descriptions did you describe anything like:

> If that fails, it either informs the user,

If you'd have led with that, then it would have been more obvious that you were including the _inform the user_ within your previous solutions.

**Comment by Artiume** [comment]
I felt that the ergonomics of the fix was understood and implied. It's very obvious that jellyfin does not behave as is expected. So I was focusing on the problem. If you feel someone is wrong, give some respect. It goes a long way even when someone is annoying you because of a misunderstanding.

**Comment by sevenrats** [comment]
+1 but i think this issue needs to be opened in respective clients.  web's handling of playback failure is insufficient.
it would have been much more efficient to leave out the mention of corrupt videos and simply discuss the error handling, but, we got here anyway.
